cmd/fetch_repo/module.go

```diff
@@ -65,10 +65,15 @@ func fetchModule(dest, importpath, version, sum string) error {
 	}
 
 	if repoSum != sum {
+		// The module cache is corrupt. Remove the downloaded directory.
+		os.RemoveAll(dl.Dir)
```

Comment:
This operation isn't atomic. Could this result in races with other concurrent runs? 

An alternative could be to move the directory into an unused location on the same file system and then delete it.

Reply:
I'm not sure if racing will be an issue: I suspect it will work. But moving then deleting could introduce other problems on Windows. Deleting or renaming files that have recently been written or renamed is unreliable.

This doesn't need to be a fatal error, but it might be worth capturing any error returned and including that in the error message.

Reply:
> Could this result in races with other concurrent runs?

It probably would, but since the Go module is already corrupted, the concurrent runs will fail anyways even without the race